libs/langchain-azure-cosmosdb/src/chat_histories.ts

```diff
@@ -0,0 +1,166 @@
+  databaseName?: string;
+  containerName?: string;
+  credentials?: TokenCredential;
+  ttl?: number;
+}
+
+/**
+ * Class for storing chat message history with Cosmos DB NoSQL. It extends the
+ * BaseListChatMessageHistory class and provides methods to get, add, and
+ * clear messages.
+ *
+ * @example
+ * ```typescript
+ * const chatHistory = new AzureCosmsosDBNoSQLChatMessageHistory({
+ *     sessionId: new ObjectId().toString(),
+ *     userId: new ObjectId().toString(),
+ *     databaseName: DATABASE_NAME,
+ *     containerName: CONTAINER_NAME,
+ * });
+ *
+ * const chain = new ConversationChain({
+ *   llm: new ChatOpenAI({ modelName: "gpt-3.5-turbo", temperature: 0 }),
+ *   memory: { chatHistory },
+ * });
+ *
+ * const response = await chain.invoke({
+ *   input: "What did I just say my name was?",
+ * });
+ * console.log({ response });
+ * ```
+ * ```
+ */
+
+export class AzureCosmsosDBNoSQLChatMessageHistory extends BaseListChatMessageHistory {
+  lc_namespace = ["langchain", "stores", "message", "azurecosmosdb"];
+
+  private container: Container;
+
+  private sessionId: string;
+
+  private databaseName: string;
+
+  private containerName: string;
+
+  private client: CosmosClient;
+
+  private userId: string;
+
+  private ttl: number | undefined;
+
+  private messageList: BaseMessage[] = [];
+
+  constructor(chatHistoryInput: AzureCosmosDBNoSQLChatMessageHistoryInput) {
+    super();
+
+    this.sessionId = chatHistoryInput.sessionId;
+    this.databaseName = chatHistoryInput.databaseName ?? DEFAULT_DATABASE_NAME;
+    this.containerName =
+      chatHistoryInput.containerName ?? DEFAULT_CONTAINER_NAME;
+    this.userId = chatHistoryInput.userId;
+    this.ttl = chatHistoryInput.ttl;
+    this.client = this.initializeClient(chatHistoryInput);
+  }
+
+  private initializeClient(
+    input: AzureCosmosDBNoSQLChatMessageHistoryInput
+  ): CosmosClient {
+    const connectionString =
+      input.connectionString ??
+      getEnvironmentVariable("AZURE_COSMOSDB_NOSQL_CONNECTION_STRING");
+    const endpoint =
+      input.endpoint ?? getEnvironmentVariable("AZURE_COSMOSDB_NOSQL_ENDPOINT");
+
+    if (!input.client && !connectionString && !endpoint) {
+      throw new Error(
+        "CosmosClient, connection string, or endpoint must be provided."
+      );
+    }
+
+    if (input.client) {
+      return input.client;
+    }
+
+    if (connectionString) {
+      const [endpointPart, keyPart] = connectionString.split(";");
+      const endpoint = endpointPart.split("=")[1];
+      const key = keyPart.split("=")[1];
+
+      return new CosmosClient({
+        endpoint,
+        key,
+        userAgentSuffix: USER_AGENT_SUFFIX,
+      });
+    } else {
+      return new CosmosClient({
+        endpoint,
+        aadCredentials: input.credentials ?? new DefaultAzureCredential(),
+        userAgentSuffix: USER_AGENT_SUFFIX,
+      } as CosmosClientOptions);
+    }
+  }
+
+  private async initializeContainer(): Promise<void> {
```

Comment:
This method is called every time messages needs to be accessed, so we don't want to make the calls to init the DB and container if not needed. Please use a boolean attribute so it's only called once.